### PR TITLE
Add "ValidatorFactory" and cleanup

### PR DESCRIPTION
**Pull request by thenotsoft**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

File: src/Validation.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+use Yiisoft\I18n\TranslatorInterface;
+use Yiisoft\Validator\Rule\Callback;
+
+final class Validation implements ValidationInterface
```
Comment (romkatsu) [inline src/Validation.php:10]: It looks like a factory. Can be named ValidatorFactory?

**Comment by terabytesoftw** [comment]
What do you think instead of putting the TranslatorInterface in the null constructor, create a setter for your configuration.

**Comment by thenotsoft** [comment]
> What do you think instead of defined the TranslatorInterface in the null constructor, create a setter for your configuration.
> 
> ```
> public function translator(TranslatorInterface $translatorInterface, string $category)
> {
> }
> ```
> 
> Locale is already defined in the interface.

I think for factory it’s more convenient in the constructor, locale is defined in Translator, but if you need translate other locale dynamic ?

File: src/Rules.php [hunk 2]
```diff
@@ -5,72 +5,35 @@
 namespace Yiisoft\Validator;
 
 use Yiisoft\Validator\Rule\Callback;
-use Yiisoft\I18n\TranslatorInterface;
 
 /**
  * Rules represents multiple rules for a single value
  */
-class Rules
+final class Rules
```
Comment (xepozz) [inline src/Rules.php:12]: Why?

Comment (samdark) [reply src/Rules.php:12]: Do you have a use-case for extending it?

**Comment by samdark** [comment]
👍

**Pull request merged**